### ISSUE TITLE
Update Dockerfile requirementes.txt to work together without warnings or extra installs

### ISSUE THREAD

**Issue by gordonwatts** [ssl-hep/ServiceX_Uproot_Transformer#8]
As a someone-important, I'd like to have the docker files as small and uncomplex as possible.

# Problem

- `requirements.txt` file, which is installed by the `Dockerfile`, contains lots of test-only items (pytest, etc.)
- `DOckerfile` contains things like `numpy` and `pandas` and `numba` which should probably be in the `requirements.txt` file or not there at all.
- Getting a warning `WARNING: 'conda clean --source-cache' is deprecated.` during docker file build.

Basically: getting down to a single source of truth for all packages, and also not installing anything extra in the docker image.
